server/gateway.go

```diff
@@ -1920,7 +1920,10 @@ func (s *Server) switchAccountToInterestMode(accName string) {
 			e = &insie{}
 			gin.gw.insim[accName] = e
 		}
-		gin.gatewaySwitchAccountToSendAllSubs(e, []byte(accName))
+		// Do it only if not already in InterestMode (or transitioning)
+		if e.mode != Transitioning && e.mode != InterestOnly {
```

Comment:
Can we only "transition" one way?

Reply:
not sure I understand... I wanted to have `if e.mode != InterestOnly` but then decided to be more explicit. Other choice would be `if e.mode == Optimistic`, but then again, if a new mode is added, and this code not updated, we would miss it.

Reply:
So if we are ask to switch but we are transitioning, and new mode is added, we might not do the send but we also might not be transitioning to InterestMode.

Reply:
yeah.. condition is wrong.. let me update this.

Reply:
ok

Reply:
well, no? I mean, we don't want to switch if we are transitioning or in interest mode, that is:
```
if !(e.mode == Transitioning || e.mode == InterestMode) {
  .. do it
}
```
which is equivalent to:
```
if e.mode != Transitioning && e.mode != InterestMode {
  .. do it
```
if we had a new mode, say NewMode but we still want to transition to InterestMode and say we are currently in Transitioning, this would still work, meaning that we would not switch because we already are doing it. What am I missing here?

Reply:
so to answer your original question, yes, we transition only from optimistic to interest mode. But for clarity, I am willing to replace with `if e.mode == Optimistic`, and will be careful to add there if we *ever* add a new mode.

Reply:
Ok, updated and push forced.